File: src/main/java/com/nuecho/genesys/cli/compare/Compare.java
package com.nuecho.genesys.cli.compare;

import com.genesyslab.platform.applicationblocks.com.IConfService;
import com.genesyslab.platform.commons.connection.ConnectionException;
import com.nuecho.genesys.cli.GenesysCli;
import java.net.URISyntaxException;
import picocli.CommandLine;

@CommandLine.Command(
  name = "compare",
  description = "Compare Genesys configuration between two instances.",
  subcommands = {CompareAgents.class}
)
public final class Compare {
  @CommandLine.ParentCommand private GenesysCli genesysCli;

  public IConfService connect() throws ConnectionException, URISyntaxException {
    return genesysCli.connect();
  }
}

File: src/main/kotlin/com/nuecho/genesys/cli/Compare.kt
package com.nuecho.genesys.cli

import com.genesyslab.platform.applicationblocks.com.IConfService
import com.genesyslab.platform.commons.connection.ConnectionException
import picocli.CommandLine
import java.net.URISyntaxException

@CommandLine.Command(name = "compare", description = ["Compare Genesys configuration between two instances."], subcommands = [(CompareAgents::class)])
class Compare : BasicCommand() {
    @CommandLine.ParentCommand
    private var genesysCli: GenesysCli? = null

    @Throws(ConnectionException::class, URISyntaxException::class)
    internal fun connect(): IConfService {
        return genesysCli!!.connect()
    }
}
